pkg/scaffold/scaffold.go

```diff
@@ -165,8 +132,8 @@ func (s *Scaffold) defaultOptions(options *input.Options) error {
 		return err
 	}
 
-	s.Project, err = LoadProjectFile(options.ProjectPath)
-	if !s.ProjectOptional && err != nil {
+	s.Config, err = config.LoadFrom(options.ProjectPath)
```

Comment:
May:
```suggestion
	s.Config, err = config.NewConfigFromFile()
```

Reply:
In this case we do need the path, so what about `NewConfigFrom(path string)`?

Reply:
See https://github.com/kubernetes-sigs/kubebuilder/pull/1325#discussion_r368285731.